### DISCUSSION TITLE
Can't read Cart Transforms via GraphQL on Plus stores

### DISCUSSION THREAD

**Discussion by ctrevorrow** [Shopify/function-examples#296] [Feedback: Cart Transform API]
I'm posting about this issue here in hopes I'll get confirmation on whether or not other people are having this issue.  I'm also hoping this may allow the information to get to someone at Shopify who might know what's going wrong and/or be able to help.

I've been having a problem since last Wednesday where I cannot read Cart Transforms at all.  On Monday evening of last week, I was able to successfully install a Cart Transform function to one of the Plus stores I work on.  This Cart Transform function was from and installed (using the GraphQL cartTransformCreate mutation) by a single-store custom app I made.  I was also able to successfully run a GraphQL query in Postman which then showed there was a Cart Transform function installed.  On Wednesday of last week, I was no longer able to run this query to read Cart Transforms.  The read query I successfully ran before and now cannot is:  
```
{
  cartTransforms(first:5) {
    nodes { id }
  }
}
```
Every time I try the read query now I receive an error saying, "Access denied for cartTransforms field. Required access: \`read_cart_transforms\` access scope. Also: The user must have products permission to view cart transform functions."

I tried running this read query through the single-store custom app and with Postman using an "App Development" custom app's credentials (formerly "Private apps".  I'm trying to differentiate between installed "Custom apps" and direct from/on the store "Custom apps").  I've even installed the "Shopify GraphiQL" app and tried running the query through there.  With all three I made sure they had read and write permissions for cart transforms and products (in addition to many other access scopes I gave).  I additionally have checked the access scopes with all three many times—to ensure read_cart_transforms is granted—by running:
```
{
  currentAppInstallation {
    accessScopes {
      handle
    }
  }
}
```
My user account has nearly all of the permissions granted and I've also had our store owner try on their account.

I've been in touch with Plus Support since last Wednesday and they seem convinced no one else is having this problem.  I mentioned I was seeing the issue through all those different ways—to give an idea of how widespread the issue is and also rule things out.  I focused on it not working in the "Shopify GraphiQL" app in subsequent communications since that would rule out problems in the setup of my app(s).  All they've been able to suggest so far is that I uninstall and re-install the "Shopify GraphiQL" app and make sure to give it the read_cart_transforms access scope, which I've done many times (including before contacting them) to no avail.

**Comment by ignacio-chiazzo** [comment]
Hi thanks for reporting. That's strange, do you have a request ID I can get? or shop_id?
